apis/v1alpha2/shared_types.go

```diff
@@ -107,10 +107,15 @@ type RouteForwardTo struct {
 	// with the "DegradedRoutes" reason. The gateway status for this route should
 	// be updated with a condition that describes the error more specifically.
 	//
+	// If there is a cross-namespace reference to an *existing* object
+	// with no ReferencePolicy, the controller must ensure the "ResolvedRefs"
+	// condition on the Gateway is set to `status: false`, with the "RefNotPermitted"
+	// reason and not configure this route in the underlying implementation.
+	//
 	// Support: Custom
 	//
 	// +optional
-	BackendRef *LocalObjectReference `json:"backendRef,omitempty"`
+	BackendRef *ObjectReference `json:"backendRef,omitempty"`
```

Comment:
This is on a collision course with GEP #718, but since there's not an implementation PR for that one yet, I think this one will likely make it in first.